### PR TITLE
Add bridges subtree

### PR DESCRIPTION
**Pull request by svyatonik**
To add a bridges subtree is the only integration option we support now:
- we don't publish on cargo;
- we are a separate repo with our own `Cargo.lock`, which will cause cargo conflicts if we'll be plugging our code using git references.

Before this PR, we have had two main branches in [our repo](https://github.com/paritytech/parity-bridges-common):
- master is used for daily development (bridges v2);
- `polkadot-staging` is what we are using to build Kusama <> Polkadot bridge. The code there is audited (wip), but it uses `polkadot-sdk` git references;
- `polkadot-v1.0.0` is a (community introduced) branch forked off master, but it references `polkadot-v1.0.0.0` branches of Substrate and Cumulus + `release-v1.0.0` branch of Polkadot.

This codebase uses `v1.0.0.0` git references, so I had to add [another branch](https://github.com/paritytech/parity-bridges-common/tree/polkadot-v.1.0.0-audited) to be able to use it here.

Once subtree is added, I'll add all required bridge pallets to the runtime + will add a relay code to our codebase. Once it is done, we could start working on local tests (zombienet based) for this bridge.

### CODE REVIEW

**Comment by svyatonik** [comment]
cc @zdave @NachoPal

**Comment by zdave-parity** [comment]
How would I update this subtree? Could you stick something in `docs`?

**Comment by svyatonik** [comment]
> How would I update this subtree? Could you stick something in `docs`?

I've added [this document](https://github.com/zdave-parity/polkadot-bulletin-chain/pull/18/commits/b5aaee1d9c1306cda67d59fafbbea36b9159630c)

**Comment by svyatonik** [comment]
Actually let me check @NachoPal suggestion to reuse bridges subtree from Cumulus repo :) Don't know if it'll work

File: BRIDGES.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# this will update new git branches from bridges repo
+# there could be unresolved conflicts, but dont worry,
+# lots of them are caused because of removed unneeded files with patch step,
+git fetch bridges --prune
+git subtree pull --prefix=bridges bridges polkadot-v.1.0.0-audited --squash
+
+# so, after fetch and before solving conflicts just run patch,
+# this will remove unneeded files and checks if subtree modules compiles
+./brides/scripts/verify-pallets-build.sh --ignore-git-state --no-revert
```
Comment (zdave-parity) [inline BRIDGES.md]: s/brides/bridges/

File: BRIDGES.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+# if there are conflicts, this could help,
+# this removes locally deleted files at least (move changes to git stash for commit)
+git status -s | awk '$1 == "DU" || $1 == "D" || $1 == "MD" || $1 == "AD" {print $2}' | grep "^brides/" | xargs git rm -q --ignore-unmatch
```
Comment (zdave-parity) [inline BRIDGES.md]: Ditto

**Comment by zdave-parity** [comment]
> Actually let me check @NachoPal suggestion to reuse bridges subtree from Cumulus repo :) Don't know if it'll work

This subtree seems preferable to me?

**Comment by NachoPal** [comment]
> > Actually let me check @NachoPal suggestion to reuse bridges subtree from Cumulus repo :) Don't know if it'll work
> 
> This subtree seems preferable to me?

Why is it necessary to copy the whole subtree instead of importing what's needed from `polkadot-sdk`?

**Comment by zdave-parity** [comment]
Well it seems that using `git subtree` is the recommended way to pull the bridges stuff into your project. Yes, it seems that Cumulus has pulled a copy in that it might be possible to reference, but adding a dependency on Cumulus just for this does not seem wise to me.

**Comment by NachoPal** [comment]
What do you think @svyatonik? What will happen for the next release? what option is easier to maintain?

**Comment by svyatonik** [comment]
I see you stick now to the `polkadot-v1.0.0`. If we'll keep doing that, then subtree is the option. If we'll migrate to `polkadot-sdk` references, then probably referencing `polkadot-sdk/bridges` is better. However I'm still not sure if this is possible.

So please decide - whether we need to migrate to `polkadot-sdk` or leave `1.0.0` refs. This should answer the other question.

> importing what's needed 

@NachoPal Actually this isn't the whole brdges repo in this subtree - [you can see](https://github.com/paritytech/parity-bridges-common) there's a lot of other crates there. We are only adding what we need. But this "what we need" is for bridge hub parachains, so it references some Cumulus stuff. We could go further and start removing everything cumulus specific from this subtree, but it'd require some more efforts. So if we start with subtree, let's start with what we already have and afterwards we'll remove extra stuff that is not required here (we need to alter or add a script similar to [this one](https://github.com/paritytech/parity-bridges-common/blob/master/scripts/verify-pallets-build.sh))

**Comment by NachoPal** [comment]
Ok, let's use subtree

**Comment by NachoPal** [comment]
> So please decide - whether we need to migrate to polkadot-sdk or leave 1.0.0 refs. This should answer the other question.

@svyatonik Actually, we should reference to `polkadot-sdk`, but the question here is what will happen with bridges stuff when `polkadot-sdk` crates are released. Then we should ref the crates, and not any `polkadot-sdk` branch. We'll have to stick to `polkadot-sdk` release branches until bridges is audited?

**Comment by svyatonik** [comment]
I think in this case we could just change the [polkadot-v.1.0.0-audited](https://github.com/paritytech/parity-bridges-common/tree/polkadot-v.1.0.0-audited) branch to reference the Substrate/Polkadot crates from `crates.io`. And keep using the subtree. We are not planning to publish bridges crates from `polkadot-sdk` in near future (until we'll resolve issues we have with our testnets + migrate to `polkadot-sdk` + deprecate the `parity-bridges-common` repo at all).

**Comment by svyatonik** [comment]
Hey @NachoPal @zdave-parity . Let's use subtree for now - we may reconsider it later. If you don't have any other questions/suggestions for this PR, let's maybe merge it? I have more follow-up PRs depending on that (still will need some time to polish the code).

**Review by NachoPal — APPROVED** [review]

**Review by zdave-parity — APPROVED** [review]

**Comment by zdave-parity** [comment]
Thought I had approved this before, sorry...

**Pull request merged**